int32 trainID
bool stopTrain
